portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,46 +49,66 @@ public void init(){
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
 
+    UserService userService = UserServiceFactory.getUserService();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    if (userService.isUserLoggedIn()) {
+        List<Comment> comments = new ArrayList<>();
 
-    List<String> comments = new ArrayList<>();
-    if (!set) { quantity = 10; }  // Default set at max of 10 comments shown upon loading screen
-    int i = 0;
-    for (Entity entity : results.asIterable()) {
-      // Limits number of comments added to the page
-      if (i < quantity){ 
-        long id = entity.getKey().getId();
-        String message = (String) entity.getProperty("comment");
-        Date timestamp = (Date) entity.getProperty("date");
-        comments.add(new Comment(id, message, timestamp));
-        i++;
-      } else {
-        break; // Exits for-loop once requested number of comments appear
-      }
-    }
+        // Add the current user to the JSON to send over to the JS file
+        comments.add(new Comment(0, null, null, userService.getCurrentUser().getEmail()));
 
-    response.setContentType("application/json");
-    String json = new Gson().toJson(comments);
-    response.getWriter().println(json);
+        if (!set) { quantity = 10; }  // Default set at max of 10 comments shown upon loading screen
+        int i = 0;
+        for (Entity entity : results.asIterable()) {
```

Comment:
Optional Style suggestion.
You can do if (comments.size() > quantity) break; to no longer have to keep track of i and simplify the if-else statements.